brave/brave-core
# Brave Ads server retry logic should backoff exponentially with a ceiling

Resolves https://github.com/brave/brave-browser/issues/7226

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:

- Confirm catalog downloads at browser startup
- Confirm catalog is downloaded periodically (~2 hours)
- Confirm retry logic for downloading catalog (use Charles Proxy to rewrite the response status code for `ads-serve.brave.com/v2/catalog` for production or `ads-serve.bravesoftware.com/v2/catalog` for staging)
- Confirm retry logic for refilling confirmation tokens (use Charles Proxy to rewrite the response status code for `GET /v1/confirmation/token/{payment_id}?nonce={nonce}`)
- Confirm success path for refilling confirmation tokens
- Confirm retry logic for redeeming confirmation tokens (use Charles Proxy to rewrite the response status code for each of `POST /v1/confirmation/{confirmation_id}/{credential}` and `GET /v1/confirmation/{confirmation_id}/paymentToken`)
- Confirm success path for redeeming confirmation tokens
status code for `PUT /v1/confirmation/payment/{payment_id}`)
- Confirm ad conversions are processed
- Confirm retry logic when updating ads rewards (use Charles Proxy to rewrite the response status code for each of `GET /v1/confirmation/payment/{payment_id}` and `GET /v2/wallet/{payment_id}/grants/ads`)
- Confirm success path when updating ads rewards

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
